### PR TITLE
Player Species Naming Convention Update

### PR DESCRIPTION
**Pull request by Skrem7**
# Document the changes in your pull request

Establishes the code and strings to permit for every player species to have names that better reflect current lore. Some of the newer lists are shorter, and some of the randomized names that can be generated will likely be goofy, but this is a start, for now. Not much more to say about the matter. Some current random generation has been tweaked and some existing string lists have been expanded.

Has been tested on a private server, the random name generation seems to work just fine.

# The New Naming Rules to Be Effected

Unless otherwise stated, human names are okay.

Lizard:
1) A verb, then an adjective or preposition, then a noun, all conjoined with dashes. Ex: Verbs-The-Nouns, Swims-Under-Bridges, Counts-Little-Numbers, Observes-Besides-Disaster
2) Draconic name: First name followed by a clan name, conjoined with a dash. Should generally sound alien, such as using a lot of vowels in a name or none at all. Not inherently restrictive, but two human names shouldn’t be conjoined. Ex: Rixitul-Usii, Vextri-Sthn, Rei-Savatum, Fuula-Ghren

Phytosian:
1) Any lizard or mothperson name is valid, in addition to human names. Ex: Vulgaris Monstrata, Deeza-Zytl, Waters-The-Lilies, Anthony Dirkin
2) The single name of any plant. A surname may be added. Ex: Cabbage, Grape Smith, Nettle, Coffee Talgot

Plasmaman:
Any noun that refers to an object followed by roman numerals. Elements from the periodic table are preferred (but not required). Ex: Gold IV, Carbon XXXII, Leopard LXI, Herald V

Mothperson:
Their names tend to be pun-related or latin-sounding in origin. First names often take latin stems and sounds, generally ending with an a, e, or s, while surnames tend to end with an a, i, or s. However, these are trends that do not need to be followed. Ex: Moffus Regalia, Cewlgaia Sanguis, Angelica Absconditi

Ethereal:
1) Any celestial body, object, or concept, followed by roman numerals. Ex: Centauri VI, Borealis LX, Sol III
2) A single word name or nickname, no surname. Ex: Pulvolis, Eteriua, Baclar

Preternis:
1) An adjective followed by a noun, typically an object. Ex: Strong Iron, Wise Brick, Vast Atlas
2) A shadow name/vxtrin name, difficult to pronounce and alien. Can string together up to three different sections using apostrophes. Ex: Krst'staun, Vxy'trmn'hul, Xhlri
3) A robotic name that can use numbers, dashes, and periods. Ex: PL-935, 6.7-RL9, R.A.Z.O.R

Polysmorph:
A single word name or nickname, potentially followed by a surname. The first name typically restricts itself to two or three syllables, ends with a vowel, and utilizes uncommon letters or letter combinations, such as "x" "v" "y" "gh" "aa" "thr", and so on. Ex: Gaathryn, Xyna, Thryv, Alei, Saavi
**Cannot use human names**

IPC:
Any valid posibrain or cyborg name, though non-acronym names should use proper capitalization of nouns (such as Spark instead of SPARK). Ex: SHIV-936, H.E.L.P.E.R, Salient 3.1, Carter 3-S50
**Cannot use human names**

# Why It's Good for the Game

We have lore (surprising, I know). Establishing newer, more unique naming conventions to reflect this lore should help players to more accurately represent the species they choose to play as well as giving examples that can be bounced off of to potentially let people make their own names. The new conventions should also loosen some more restrictive rules at the moment, especially for IPCs. The expansion of polymorph names should also help to reduce repeat of names, though I can't promise it won't still happen, considering that their name generation is based off of a single string with no potential variation in including another file.

# Wiki Documentation

If council passes this, the naming convention page on the wiki will have to be rectified. Species-specific pages should likely have their naming conventions updated as well

# Changelog

:cl:  
rscadd: Phytosians, preterni, and ethereal all have new naming schema regarding randomized names
rscadd: Existing lists such as polymorphs and the old ethereal list (which was based on constellations) have been expanded
tweak: Lizard name generation has been tweaked to reflect current lore better
/:cl:

### CODE REVIEW

**Comment by RG4ORDR** [comment]
Add Bingles to the Polymorph name, ty

**Comment by Moltijoe** [comment]
Looking forward to seeing more vxtrin names. Gonna be a nightmare trying to tell multiple preterni with them apart.

**Review by Moltijoe — APPROVED** [review]
Code checks out
looking forward to seeing more Vxtrin names

File: code/__HELPERS/names.dm [hunk 1]
```diff
@@ -1,17 +1,50 @@
...
+/proc/preternis_name(gender)
+	var/choice = rand(1,3)
+	if(choice == 1) //Adj + Noun
+		return "[pick(GLOB.adjectives)] [pick(GLOB.forge_name)]"
+	if(choice == 2) //Vxtrin name
+		if(prob(25)) //Single Word
+			return pick(GLOB.preternis_names)
+		if(prob(50)) //Caste + Name
+			return "[pick(GLOB.preternis_class)]'[pick(GLOB.preternis_names)]"
+		return "[pick(GLOB.preternis_class)]'[pick(GLOB.preternis_names)]'[pick(GLOB.preternis_home)]" //Caste + Name + Home
```
Comment (JamieD1) [inline code/__HELPERS/names.dm:36]:
```suggestion
	var/choice = rand(1,4)
	if(choice == 1) //Human
		if(gender == MALE)
			randname = pick(GLOB.first_names_male)
		else
			randname = pick(GLOB.first_names_female)
		randname += " [pick(GLOB.last_names)]"
		return randname
	if(choice == 2) //Lizard
		return lizard_name(gender)
	if(choice == 3) //Moth
		return moth_name()
	if(choice == 4) //Plant
		randname = pick(GLOB.plant_names)
		if(prob(25))
			randname += " [pick(GLOB.last_names)]"
		return randname

/proc/preternis_name(gender)
	var/choice = rand(1,3)
	if(choice == 1) //Adj + Noun
		return "[pick(GLOB.adjectives)] [pick(GLOB.forge_name)]"
	if(choice == 2) //Vxtrin name
		if(prob(25)) //Single Word
			return pick(GLOB.preternis_names)
		if(prob(50)) //Caste + Name
			return "[pick(GLOB.preternis_class)]'[pick(GLOB.preternis_names)]"
		return "[pick(GLOB.preternis_class)]'[pick(GLOB.preternis_names)]'[pick(GLOB.preternis_home)]" //Caste + Name + Home
		
		
		
		AHHHH USE SWITCH NOT YANDERE DEV STYLE
```

**Review by JamieD1 — CHANGES_REQUESTED** [review]
AHHHH SO MANY IF CHOICES,

MAKE SWITCHES INSTEAD DONT DO A YANDERE DEV

**Comment by adamsong** [comment]
@JamieD1 This one good?

File: yogstation/code/modules/mob/living/carbon/human/species_types/plantpeople.dm [hunk 2]
```diff
@@ -282,4 +282,12 @@
 		if(/obj/item/projectile/energy/florayield)
 			H.nutrition = min(H.nutrition+30, NUTRITION_LEVEL_FULL)
 
+/datum/species/pod/random_name(gender,unique,lastname)
+	if(unique)
+		return random_unique_pod_name(gender)
+
+	var/randname = pod_name(gender)
+
+	return randname
```
Comment (JamieD1) [inline yogstation/code/modules/mob/living/carbon/human/species_types/plantpeople.dm]:
```suggestion
	if(unique)
		return random_unique_pod_name(gender)
	return pod_name(gender)
```

File: yogstation/code/modules/mob/living/carbon/human/species_types/preternis/preternis.dm [hunk 3]
```diff
@@ -273,3 +273,11 @@ adjust_charge - take a positive or negative value to adjust the charge level
...
 		return 1
 	return 0
+
+/datum/species/preternis/random_name(gender,unique,lastname)
+	if(unique)
+		return random_unique_preternis_name()
+
+	var/randname = preternis_name()
+
+	return randname
```
Comment (JamieD1) [inline yogstation/code/modules/mob/living/carbon/human/species_types/preternis/preternis.dm]:
```suggestion
/datum/species/preternis/random_name(gender,unique,lastname)
	if(unique)
		return random_unique_preternis_name()
	return preternis_name()
```

File: code/__HELPERS/names.dm [hunk 4]
```diff
@@ -1,17 +1,52 @@
...
 
 /proc/ethereal_name()
-	var/tempname = "[pick(GLOB.ethereal_names)] [random_capital_letter()]"
-	if(prob(65))
-		tempname += random_capital_letter()
-	return tempname
+	if(prob(66))
+		return "[pick(GLOB.constellations)] \Roman[rand(1,99)]"
+	else
+		return "[pick(GLOB.ethereal_names)]"
```
Comment (JamieD1) [inline code/__HELPERS/names.dm]:
```suggestion
	if(prob(66))
		return "[pick(GLOB.constellations)] \Roman[rand(1,99)]"
	return "[pick(GLOB.ethereal_names)]"
```

File: code/__HELPERS/names.dm [hunk 5]
```diff
@@ -1,17 +1,52 @@
 /proc/lizard_name(gender)
 	if(gender == MALE)
-		return "[pick(GLOB.lizard_names_male)]-[pick(GLOB.lizard_names_male)]"
+		return "[pick(GLOB.lizard_names_male)]-[pick(GLOB.lizard_names_clan)]"
 	else
```
Comment (JamieD1) [inline code/__HELPERS/names.dm]:
```suggestion
```

File: code/__HELPERS/names.dm [hunk 6]
```diff
@@ -1,17 +1,52 @@
...
-		return "[pick(GLOB.lizard_names_female)]-[pick(GLOB.lizard_names_female)]"
+		return "[pick(GLOB.lizard_names_female)]-[pick(GLOB.lizard_names_clan)]"
```
Comment (JamieD1) [inline code/__HELPERS/names.dm]:
```suggestion
	return "[pick(GLOB.lizard_names_female)]-[pick(GLOB.lizard_names_clan)]"
```

**Review by JamieD1 — CHANGES_REQUESTED** [review]
This might make it more efficient

**Comment by Moltijoe** [comment]
> Code checks out

Jamie proceeds to find 19 quadrillion problems

I'm a maintainer I swear

**Comment by JamieD1** [comment]
If you return you dont need an else as its the end of the proc

**Pull request merged**